src/components/ComponentList.js

```diff
@@ -164,13 +166,24 @@ export default class ComponentList extends React.Component {
   }
 
   render() {
-    const { loadMoreRows, listHeight, noRowsRenderer, list, fetchingRenderer, renderFilterBar } = this.props
+    const {
+      loadMoreRows,
+      listHeight,
+      noRowsRenderer,
+      list,
+      isFetching,
```

Comment:
Looking at this now we could likely generalize a little. Rather than passing in a `noRowsRender`, a `fetchingRenderer`, and an `isFetching` flag, we can pass in a well-configured `noRowsRenderer`. We should likely make a new component (e.g., EmptyListRenderer or some such) that looks a lot like the `wrappedNoRowsRender` in `RowEntityList`. For example,

```javascript
  static defaultProps = {
    fetchingRenderer: () => <div>Loading...</div>,
    noRowsRenderer: () => <div>Nothing found</div>
  }
  render() {
    const { list } = this.props
    const renderer = list.isFetching ? fetchingRenderer : noRowsRenderer
    return <div className={'list-noRows'}>{renderer()}</div>
  }
```

This would be configured and passed in in the `render` of `PageDefinitions`

Reply:
I am not understanding this and would like more explanation when you have time.